gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/promotion/model/PromotionQuery.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.core.promotion.model;
+
+import java.util.Set;
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+public class PromotionQuery {
```

Comment:
what about using a Record for `PromotionQuery` and `PromotionRequest` (even PromotionAuthor` and `Promition`?

We should favor record now it is available that lombok @Data which result the same 🤔